### ISSUE TITLE
How to debug `AMQJS0008I Socket closed.`. wasting 3 days

### ISSUE THREAD

**Issue by matart15** [aws-amplify/amplify-js#9583]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

PubSub

### Amplify Categories

auth

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

AMQJS0008I Socket closed.

1. iot mqtt test works with cert keys.
2. did everythong described [here](https://docs.amplify.aws/lib/pubsub/getting-started/q/platform/js/#step-2-attach-your-policy-to-your-amazon-cognito-identity) and [here](https://github.com/aws-amplify/docs/issues/460#issuecomment-481499155)
3. Checked IAM policy like hundred times.
4. Checked IOT policy attach like thousand times
5. checked core version
```
amplified_todo@0.0.1
├── @aws-amplify/core@4.3.14
└─┬ aws-amplify@4.3.14
  ├─┬ @aws-amplify/analytics@5.1.12
  │ └── @aws-amplify/core@4.3.14 deduped
  ├─┬ @aws-amplify/api@4.0.32
  │ ├─┬ @aws-amplify/api-graphql@2.2.21
  │ │ └── @aws-amplify/core@4.3.14 deduped
  │ └─┬ @aws-amplify/api-rest@2.0.32
  │   └── @aws-amplify/core@4.3.14 deduped
  ├─┬ @aws-amplify/auth@4.4.1
  │ └── @aws-amplify/core@4.3.14 deduped
  ├─┬ @aws-amplify/cache@4.0.34
  │ └── @aws-amplify/core@4.3.14 deduped
  ├── @aws-amplify/core@4.3.14 deduped
  ├─┬ @aws-amplify/datastore@3.7.6
  │ └── @aws-amplify/core@4.3.14 deduped
  ├─┬ @aws-amplify/geo@1.2.0
  │ └── @aws-amplify/core@4.3.14 deduped
  ├─┬ @aws-amplify/interactions@4.0.32
  │ └── @aws-amplify/core@4.3.14 deduped
  ├─┬ @aws-amplify/predictions@4.0.32
  │ └── @aws-amplify/core@4.3.14 deduped
  ├─┬ @aws-amplify/pubsub@4.2.8
  │ └── @aws-amplify/core@4.3.14 deduped
  ├─┬ @aws-amplify/storage@4.4.15
  │ └── @aws-amplify/core@4.3.14 deduped
  └─┬ @aws-amplify/xr@3.0.32
    └── @aws-amplify/core@4.3.14 deduped
```

Stupid error only displays `AMQJS0008I Socket closed.`

### Expected behavior

just publish something

### Reproduction steps

https://docs.amplify.aws/lib/pubsub/getting-started/q/platform/js/

# HOW DO I DEBUG THIS?

**Comment by matart15** [comment]
I thought it got fixed. but still happening again

**Comment by stocaaro** [comment]
Hello @matart15,

Reading over your notes, I've walked over these same tutorials and gotten the applications working. I've reconstructed this base application a number of times and have certainly tripped into cases where it fails and when its not working it can be a challenge to determine why.

Do you have an example repo that is having this issue that I could checkout the code on? Alternatively, have you tried increasing the log level to get more information in the browser console? 

To do so, add this after the `Amplify.configure` call.
```
Amplify.Logger.LOG_LEVEL = 'DEBUG';
```

Hopefully you've seen the other side of this; if not, can you provide more information so I can help you get unblocked?

Thanks,
Aaron

**Comment by matart15** [comment]
@stocaaro 

Thank you for looking at it.
And sorry I forgot to mention, AWS team fixed it.

I sent CUSTOMER support ticket and they said it is fixed can you try again. And it worked.

I have no idea what went wrong. and I asked them how to debug same things, they did not respond. I moved on.

So, I assume it was some bug on AWS side.

thank you again for taking time.

Closing the ussue

**Comment by david-mcafee** [comment]
Reopening, as I'm encountering this as well. I have [an Amplify demo app with a simple PubSub component](https://github.com/david-mcafee/amplify-demo/blob/main/src/Components/PubSub/index.tsx) that was working previously. The only update I've made was to update the Amplify version to latest.

**Comment by cwomack** [comment]
Hello @matart15, between your above [comment](https://github.com/aws-amplify/amplify-js/issues/9583#issuecomment-1129765672) regarding the AWS customer support ticket stating it's fixed, reviewing it with @david-mcafee internally, and attempting to reproduce the bug myself (with no success) on the most version of amplify... I'll close this issue out again.  

Anyone in the community can feel free to comment if this is still being experienced and we can re-open the issue if needed.

**Comment by BTsykaniuk** [comment]
Hi! We also ran into a problem connecting to the pub.sub. We made the settings in accordance with the documentation: we added all the necessary permissions for roles, we add users to the policy. But when the web client tries to subscribe to a topic, we get the following error
![image (4)](https://user-images.githubusercontent.com/41420987/217237061-4ee600e2-14e0-43ad-aa28-81d6329393a1.png)
The code we will use on the frontend
`Amplify.addPluggable(
  new AWSIoTProvider({
    aws_pubsub_region: 'us-east-1',
    aws_pubsub_endpoint: 'wss://a1exwp3w8uqoui-ats.iot.us-east-1.amazonaws.com/mqtt',
  })
);

Amplify.configure({
  ...awsExports,
  Auth: {
    identityPoolId: 'us-east-1:8da61e86-1afd-407c-8c94-3d5883dccede',
    userPoolId: 'us-east-1_exv7kQbdD',
    region: 'us-east-1',
  }
});

Hub.listen('pubsub', (data) => {
  const { payload } = data;
  if (payload.event === CONNECTION_STATE_CHANGE) {
    const { connectionState } = payload.data;
    console.log(connectionState, data);
  }
});

const usePubSubSubscription = () => {
  const { user } = useAuthenticator((context) => [
    context.user,
  ]);

  useEffect(() => {
    if (!user) {
      return undefined;
    }
    const sub = PubSub.subscribe('myIoTPolicy').subscribe({
      next: (data) => console.log('data', data),
      error: (errorValue) => console.log('error', errorValue),
      complete: () => console.log('done'),
    });
    return () => {
      sub.unsubscribe();
    };
  }, [user]);
};`

`@aws-amplify/pubsub 5.0.12
aws-amplify 5.0.12
@aws-amplify/ui-react 4.3.6`

We will be very grateful for your help!

**Comment by stocaaro** [comment]
Hello @BTsykaniuk,

The issue you are describing sounds different than the one encountered by the original requestor. Rather than re-opening this issue, I think it would be clearer to troubleshoot the problem your seeing as a [new issue](https://github.com/aws-amplify/amplify-js/issues/new?assignees=&labels=pending-triage&template=1.bug_report.yaml).

The error you're seeing looks like the error I've encountered when I'm setting up an IoT connection for the first time and haven't got the permissions quite right. Have you seen this work for this app/backend or are you setting this up for the first time?

If this is the first time, please go through and double check the the IAM/Policy config you mentioned and make sure everything lines up with [the PubSub getting started guide](https://docs.amplify.aws/lib/pubsub/getting-started/q/platform/js/) directions.

Things I would double check first:
- Does the `AuthRole` have both granted IoT Access policies attached to it?
- Does the IoT security policy show the `identityId` of your user in the Targets list in the IoT console?
- Can you confirm that the `Policy resource` on your IoT policy has your account number and region formatted correctly?

I've been through these steps quite a few times and have found that small mistakes error out in the same way as having done no configuration at all. There's no feedback until it finally works and troubleshooting the configuration is a bit tedious.

I hope something in this helps you get unblocked. If you're still having trouble with this, please open a [new issue](https://github.com/aws-amplify/amplify-js/issues/new?assignees=&labels=pending-triage&template=1.bug_report.yaml) with all of this detail and anything further you've discovered.

Thanks!

**Comment by BTsykaniuk** [comment]
@stocaaro yes, this was some issue with `Policy resource`. Now all works fine! Thank you for support!

**Comment by paulinesarana19** [comment]
Hi @cwomack , having an inconsistent issue with this error. We are getting this error on some certain periods of the day only(around the morning our time (GMT+8)), but is able to connect and subscribe to topics around noon or after lunch our time. Any idea how to resolve this? Thanks!
![image](https://github.com/aws-amplify/amplify-js/assets/92791096/e3313a89-c0e3-400c-aa7f-4b6a1de37c1c)
